aws-sns-subscription/src/main/java/software/amazon/sns/subscription/UpdateHandler.java

```diff
@@ -21,90 +25,88 @@ protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            .then(progress -> checkSubscriptionExistsAndSetSubscriptionArn(proxy, proxyClient, request, progress, callbackContext, false, logger))
+            .then(progress -> validateRegionUpdate(previousModel, currentModel, request, progress))
+            .then(progress -> modifyString(proxy, proxyClient, currentModel.getFilterPolicyScope(), currentModel, SubscriptionAttribute.FilterPolicyScope, previousModel.getFilterPolicyScope(), progress, logger))
             .then(progress -> modifyPolicy(proxy, proxyClient, currentModel.getFilterPolicy(), currentModel, SubscriptionAttribute.FilterPolicy, previousModel.getFilterPolicy(), progress, logger))
             .then(progress -> modifyPolicy(proxy, proxyClient, currentModel.getDeliveryPolicy(), currentModel, SubscriptionAttribute.DeliveryPolicy,previousModel.getDeliveryPolicy(), progress, logger))
             .then(progress -> modifyPolicy(proxy, proxyClient, currentModel.getRedrivePolicy(), currentModel, SubscriptionAttribute.RedrivePolicy,previousModel.getRedrivePolicy(), progress, logger))
-            .then(progress -> modifyRawMessageDelivery(proxy, proxyClient, currentModel.getRawMessageDelivery(), currentModel, SubscriptionAttribute.RawMessageDelivery,previousModel.getRawMessageDelivery(), progress, logger))
-            .then(progress -> modifySubscriptionRoleArn(proxy, proxyClient, currentModel.getSubscriptionRoleArn(), currentModel, SubscriptionAttribute.SubscriptionRoleArn, previousModel.getSubscriptionRoleArn(), progress, logger))
-            .then(progress -> new ReadHandler().handleRequest(proxy, request, callbackContext, proxyClient, logger));
+            .then(progress -> modifyBoolean(proxy, proxyClient, currentModel.getRawMessageDelivery(), currentModel, SubscriptionAttribute.RawMessageDelivery,previousModel.getRawMessageDelivery(), progress, logger))
+            .then(progress -> modifyString(proxy, proxyClient, currentModel.getSubscriptionRoleArn(), currentModel, SubscriptionAttribute.SubscriptionRoleArn, previousModel.getSubscriptionRoleArn(), progress, logger))
+            .then(progress -> ProgressEvent.defaultSuccessHandler(progress.getResourceModel()));
     }
 
-    private ProgressEvent<ResourceModel, CallbackContext> modifySubscriptionRoleArn(
+    private ProgressEvent<ResourceModel, CallbackContext> validateRegionUpdate(ResourceModel previousModel, ResourceModel currentModel, ResourceHandlerRequest<ResourceModel> request, ProgressEvent<ResourceModel, CallbackContext> progress) {
+        String prevRegion = previousModel.getRegion();
+        String curRegion = currentModel.getRegion();
+        final String currentRequestRegion = request.getRegion();
+
+        if (!StringUtils.equals(
+                prevRegion == null ? currentRequestRegion : prevRegion,
+                curRegion == null ? currentRequestRegion : curRegion)) {
+            throw new CfnNotUpdatableException(ResourceModel.TYPE_NAME, currentModel.getRegion());
```

Comment:
is this the same as throwing a Terminal Exception? https://code.amazon.com/packages/AWS21Workflows/blobs/mainline/--/src/com/amazon/aws21/workflows/resources/sns/subscription/SubscriptionUpdateWorkflowCustomImpl.java#:~:text=getRegion()))%20%7B-,throw%20new%20TerminalException(,().getRegion()%2C%20subscriptionArn.getRegion())%2C%20FailureMode.UNTOUCHED)%3B,-%7D